src/main/java/org/luksza/gerrit/rest/PostRetrigger.java

```diff
@@ -88,10 +104,36 @@ public Object apply(ChangeResource resource, Input input)
 
   private static final String STATUS_OPEN = "status:open";
 
+  private Optional<Header> getAuth(String jenkinsUrl) {
+    String userName = config.getJenkinsUserName(jenkinsUrl);
+    String token = config.getJenkinsToken(jenkinsUrl);
+    if (userName == null || token == null) {
+      return Optional.absent();
+    }
 
-  private Optional<String> createSession(HttpClient client, Project.NameKey project)
-      throws UnsupportedEncodingException, IOException, ClientProtocolException {
-    HttpPost search = prepareFormPost(urls.getSearchUrl(project));
+    String auth = userName + ":" + token;
+    byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(Charset.forName("ISO-8859-1")));
+    Header header = new BasicHeader(HttpHeaders.AUTHORIZATION, "Basic " + new String(encodedAuth));
+    return Optional.of(header);
+  }
+
+  private Optional<Header> getCrumb(HttpClient client, Project.NameKey project, Optional<Header> auth) {
```

Comment:
I would consider using BasicHeader as a Optional parameter type and don't do casting line 127. After all this is a private API, so we can use more specific types